cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceQueryAdmin.java

```diff
@@ -56,7 +58,8 @@ public DefaultNamespaceQueryAdmin(NamespaceStore nsStore) {
    * @throws NamespaceNotFoundException if the requested namespace is not found
    */
   @Override
-  public NamespaceMeta get(Id.Namespace namespaceId) throws Exception {
+  public NamespaceMeta get(Id.Namespace namespaceId) throws NamespaceNotFoundException, IOException,
+    UnauthenticatedException {
```

Comment:
@anwar6953 : That change is needed for impersonation right ? Lets do it in an independent PR.